app/services/foreman/provision/ssh.rb

```diff
@@ -50,64 +71,72 @@ def command_prefix
     (username == "root") ? "" : "sudo "
   end
 
-  def command
-    # Use the users home to store the provision script since we can't reliably
-    # tell if other locations are writeable or executable by the user.
+  # Use the users home to store the provision script since we can't reliably
+  # tell if other locations are writeable or executable by the user.
+  def main_command
     main_execution = "(chmod 0701 ./#{remote_script} && #{command_prefix} ./#{remote_script} ; echo $? >#{remote_script}.status) 2>&1"
-    "#{command_prefix} sh -c '#{main_execution} | tee #{remote_script}.log; exit $(cat #{remote_script}.status)'"
-  end
-
-  def defaults
-    {
-      :keys_only    => true,
-      :config       => false,
-      :auth_methods => %w(publickey),
-      :compression  => true,
-      :logger       => logger,
-    }
+    cleanup = "#{command_prefix} rm -f #{remote_script}"
+    "#{command_prefix} sh -c '#{main_execution} | tee #{remote_script}.log; #{cleanup}; exit $(cat #{remote_script}.status)'"
   end
 
   def logger
     Rails.logger
   end
 
-  def initiate_connection!
-    Timeout.timeout(Setting[:ssh_timeout].to_i) do
-      Timeout.timeout(8) do
-        ssh.run('pwd')
-      end
-    rescue Errno::ECONNREFUSED
-      logger.debug "Connection refused for #{address}, retrying"
-      sleep(2)
-      retry
-    rescue Errno::EHOSTUNREACH
-      logger.debug "Host unreachable for #{address}, retrying"
-      sleep(2)
-      retry
-    rescue Net::SSH::Disconnect
-      logger.debug "Host dropping connections for #{address}, retrying"
-      sleep(2)
-      retry
-    rescue Net::SSH::ConnectionTimeout
-      logger.debug "Host timed out for #{address}, retrying"
-      sleep(2)
-      retry
-    rescue Net::SSH::AuthenticationFailed
-      logger.debug "Auth failed for #{username} at #{address}, retrying"
-      sleep(2)
-      retry
-    rescue Timeout::Error
-      retry
-    rescue => e
-      Foreman::Logging.exception("SSH error", e)
-    end
+  def scp
+    logger.debug "about to upload #{template} to remote system at #{remote_script}"
```

Comment:
Did you intend to use `remote_host` instead of `remote_script`? And for logging a more common form is using the `-ing` form of verbs to show it's happening. Consider using this:
```suggestion
    logger.debug "Uploading template '#{template}' to remote system '#{remote_host}'"
```